test/src/main/java/org/apache/accumulo/test/fate/accumulo/AccumuloStoreReadWriteIT.java

```diff
@@ -127,6 +127,31 @@ public void testReadWrite() throws Exception {
     }
   }
 
+  @Test
+  public void testReadWriteTxInfo() throws Exception {
+    final String table = getUniqueNames(1)[0];
+    try (ClientContext client =
+        (ClientContext) Accumulo.newClient().from(getClientProps()).build()) {
+      client.tableOperations().create(table);
+
+      AccumuloStore<TestEnv> store = new AccumuloStore<>(client, table);
+
+      long tid = store.create();
+      FateTxStore<TestEnv> txStore = store.reserve(tid);
+
+      try {
+        // Go through all enum values to verify each TxInfo type will be properly
+        // written and read from the store
+        for (TxInfo txInfo : TxInfo.values()) {
+          txStore.setTransactionInfo(txInfo, "value: " + txInfo.name());
```

Comment:
Could add a check before setting to check that case.

```suggestion
          assertNull(txStore.getTransactionInfo(txInfo));
          txStore.setTransactionInfo(txInfo, "value: " + txInfo.name());
```